modules/tinymce/src/core/main/ts/delete/InlineFormatDelete.ts

```diff
@@ -47,9 +58,63 @@ const deleteCaret = (editor: Editor, forward: boolean): Optional<() => void> =>
   });
 };
 
+const isBrInEmptyElement = (elm: Element): boolean => {
+  const parentElm = elm.parentElement;
+  return !Type.isNull(parentElm) && Empty.isEmpty(SugarElement.fromDom(parentElm)) && NodeType.isBr(elm);
+};
+
+const isEmptyCaret = (elm: Element): boolean =>
+  CaretFormat.isEmptyCaretFormatElement(SugarElement.fromDom(elm));
+
+const createCaretFormatAtStart = (editor: Editor, formatNodes: Node[]): void => {
+  const startElm = editor.selection.getStart();
+  // replace <br> in empty node or existing caret at start if applicable
+  // otherwise create new caret format at start
+  const pos = isBrInEmptyElement(startElm) || isEmptyCaret(startElm)
+    ? CaretFormat.replaceWithCaretFormat(startElm, formatNodes)
+    : CaretFormat.createCaretFormatAtStart(editor, formatNodes);
+  editor.selection.setRng(pos.toRange());
+};
+
+const updateCaretFormat = (editor: Editor, updateFormats: Node[]): void => {
+  const missingFormats = Arr.difference(updateFormats, getFormatNodesAtStart(editor));
+  if (missingFormats.length > 0) {
+    createCaretFormatAtStart(editor, missingFormats);
+  }
+};
+
+const isSelectionAtStartOfContainer = (editor: Editor): boolean =>
+  editor.selection.getRng().startOffset === 0;
+
+const deleteRange = (editor: Editor): Optional<() => void> => {
+  if (isSelectionAtStartOfContainer(editor)) {
+    const formatNodes = getFormatNodesAtStart(editor);
+    return Optional.some(() => {
+      DeleteUtils.execNativeDeleteCommand(editor);
+      updateCaretFormat(editor, formatNodes);
+    });
+  } else {
+    return Optional.none();
+  }
+};
+
 const backspaceDelete = (editor: Editor, forward: boolean): Optional<() => void> =>
-  editor.selection.isCollapsed() ? deleteCaret(editor, forward) : Optional.none();
+  editor.selection.isCollapsed() ? deleteCaret(editor, forward) : deleteRange(editor);
+
+const hasAncestorInlineCaret = (elm: SugarElement<Node>): boolean =>
+  PredicateFilter.ancestors(elm, (node) => isCaretNode(node.dom), (node) => ElementType.isBlock(node)).length > 0;
```

Comment:
ETA reduce and use PredicateExists instead since you are only interested if it exists or not.

```suggestion
  PredicateExists.ancestor(elm, (node) => isCaretNode(node.dom), ElementType.isBlock);
```